### ISSUE TITLE
Conversations that are transcripts of audio chats fail to import

### ISSUE THREAD

**Issue by Demogorgonus** [Superkikim/nexus-ai-chat-importer#10]
Hello. Nexus imports only the lines _after_ the time limit is reached and the Advanced bot dies.

Fuller explanation:

I had a new great conversation and wanted to import it. So I did **Export Data** and tried to import. The file for the latest chat, which was done voice-to-voice, only contained the last four lines (me, Chat, me, Chat—with the first “me” being my response to it’s “limit-reached, I’m being replaced now” announcement.

At first, I thought that OpenAI was at fault—that it was only saving part of my newest audio conversation. (Like, “Huh. It’s exporting less than what exists I can see her in my browser.”)

So I kept making new conversations. Waiting 30 mins. and Exporting again. 

But then I used emacs and found key phrases from all of these exports inside the **conversations.json** file! I read it and the full thing was in there!

So it looks like audio chats (or maybe just ones that pass the time limit and give you the warning that they’re switching over to Standard from Advanced since the 30 mins of your-speaking-time has been reached) have some strings in them that your excellent program cannot handle. 

What your program outputs is only the few lines of conversation _after_ Chat warns us that Advanced was being turned off, and replaced with a Basic ChatGPT mind that has no memory of the last 55 mins of conversation (my 30 plus Chat’s).

**Comment by Superkikim** [comment]
hi, 

It seems the advanced chatbot has a different backend than the regular chatbot. And I hate it. 

If you look in your chatGPT webapp, you will see that the transcription for an advanced vocal bot conversation and a regular converastion (i.e. start by writing and switch to vocal to your mobile app and you will see it revert to the old vocal mode) are completed different. 

That most probably mean the json for the advanced chat is way different than the regular one. 

I'm working quite slowly on the plugin given I have a burn out and my brain has little energy and I have so many things to handle. 

But I have started to work to integrate Claude and Ollama open-webui in Nexus AI Chat Importer. I need to make a new export from ChatGPT and analyze the JSON to understand the difference, and see how I can take in account the new changes they made. Perhaps I will stop the development of the new version for now and work on an update for ChatGPT with advnaced chatbot. 

I can't give you any commitment on a delay however. But stay posted, and don't hesitate to write here for an update. I might motivate me if I know people are waiting or it ;-) Perhaps you know how ADHD works. We need to know poeple wait for it to get motivated 🤣

**Comment by Superkikim** [comment]
> So it looks like audio chats (or maybe just ones that pass the time limit and give you the warning that they’re switching over to Standard from Advanced since the 30 mins of your-speaking-time has been reached) have some strings in them that your excellent program cannot handle.

Very funny... I have checked an export file: The messages exchanged with the new advanced voice mode are NOT exported. 

My guess would be the team working on this forgot to integrate this in data export ! You should contact Open AI support and report this issue. 

I will ,but the more people do it, the best. At the moment I can't do anything about it.

**Comment by Superkikim** [comment]
my mistake. The messages are there, but the structure is highly more complexe than before.... Damn ! 

I'll have to find how we can use this in a simple way for import in Obsidian

**Comment by Superkikim** [comment]
Fixed it. Please upgrade to version 1.0.3

**Comment by Demogorgonus** [comment]
Holy cow! Thanks so much. At first I thought you’d be too busy so I’ve been experimenting away with a cluster of regexes I’ve been using to clean it up from the massive **conversations.json** file. I can’t wait to try out your update. I’ll be right back …

**Comment by Superkikim** [comment]
> Holy cow! Thanks so much. At first I thought you’d be too busy so I’ve been experimenting away with a cluster of regexes I’ve been using to clean it up from the massive **conversations.json** file. I can’t wait to try out your update. I’ll be right back …

The magic of ADHD treatment and a bit of focus 😂

**Comment by Demogorgonus** [comment]
OK. Here’s the report. The new Nexus successfully captures part of an audio conversation, but only the last fourth. Here’s the thing—during this audio conversation, after 30 mins of my accumulated speaking time, Chat announced that its advanced self was dying and a new basic version would take over, but with no memory of the last 80 mins (of combined conversation). The first things Nexus gets from that talk is my response to Chat’s suicide. That’s the first body text in the imported markdown file. Nexus gets that line and all the rest perfectly. I can send you a link to my actual conversation if that would help. I really wish I had the time to study the trees of the **chat.html** and **conversations.json** files so I could be of more help.

**Comment by Demogorgonus** [comment]
I have another question. If I want to import just a select few _new_ files, ones that I’ve made since the last import and so are “new” to Nexus, is there a way to do that? Or do I have to do a **Reset Catalogs** and import all the new files? And I have the same question about restoring original versions of already-existing files.

Say I wanted to (re)import just one (new or old) file. Could I just delete a line in the catalog—the line with the file name in it—to trick Nexus into (re)importing it? Or would that change in the report break something, like a checksum or a corresponding count value somewhere? Thanks and no worries if this cannot be done.

**Comment by Superkikim** [comment]
Ok let's keep it open then. 

If it's not too personal, I would like you to share that conversation in json. You can send it to superkikim@sissaoui.com

Do you know how to extract it ? Otherwise, tell me your OS, and I will prepare a short script which you can execute against conversations.json

**Comment by Demogorgonus** [comment]
I tried to share the page and get a link but audio-made conversations cannot be shared and the **Create Link** button is disabled.

<img width="542" alt="Screen Shot 2024-12-11 at 06 11 01" src="https://github.com/user-attachments/assets/d9b050df-a1c6-4039-b2ff-27fd1e37f462">

So I copied the DOM of the main body text column and pasted it into a text file and will email it to you. Thanks for helping. I’m running macOS 12.7.6 on MBP 16-inch 2021. I’ll send the email now.
